backend/src/layer/data/flickr_api/flickr_api_handler.rs

```diff
@@ -18,35 +18,66 @@ pub struct FlickrApiHandler {
 
 lazy_static! {
     static ref LONG_URL_REGEX: Regex =
-        Regex::new(r"(https://www.flickr.com/photos/)(\w+)/(\d+)([/]{0,1})")
+        Regex::new(r"(https://www.flickr.com/photos/)([\d\w@]+)/(\d+)([/]{0,1})")
             .expect("regex creation failed");
     static ref SHORT_URL_REGEX: Regex =
         Regex::new(r"(https://flic.kr/p/)([\d\w]+)").expect("regex creation failed");
 }
 
+const ALPHABET: &str = "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ";
+
 impl FlickrApiHandler {
     #[must_use]
-    #[allow(clippy::missing_const_for_fn)] // cannot be made const because of compiler error`
+    #[allow(clippy::missing_const_for_fn)]
     pub fn new(info: FlickrInfo) -> Self {
         Self {
             request: ApiRequest::new(info.api_key),
         }
     }
 
-    // URL TYPE 1: https://www.flickr.com/photos/gerdavs/52310534489/ <- remove last '/'
+    // URL TYPE 1.1: https://www.flickr.com/photos/gerdavs/52310534489/ <- remove last '/'
+    // URL TYPE 1.2: https://www.flickr.com/photos/198319418@N06/53077317043 <- remove last '/'
     // URL TYPE 2: https://flic.kr/p/2oRguN3
     // Both cases: Split with '/' and get last member (= photo_id).
-    fn determine_photo_id(mut url: &str) -> Result<&str> {
-        if !SHORT_URL_REGEX.is_match(url) && !LONG_URL_REGEX.is_match(url) {
-            return Err(ImageHosterError::FormatNotFound(format!(
+    fn determine_photo_id(url: &str) -> Result<String> {
+        if LONG_URL_REGEX.is_match(url) || SHORT_URL_REGEX.is_match(url) {
+            let id = url
+                .trim_end_matches('/')
+                .split('/')
+                .last()
+                .map(String::from)
+                .ok_or_else(|| {
+                    ImageHosterError::FormatNotFound(format!(
+                        "this url format is not supported: '{url}'"
+                    ))
+                })?;
+            if SHORT_URL_REGEX.is_match(url) {
+                Self::decode(&id)
+            } else {
+                Ok(id)
+            }
+        } else {
+            Err(ImageHosterError::FormatNotFound(format!(
                 "this url format is not supported: '{url}'"
-            )));
+            )))
+        }
```

Comment:
Hear me out:
bs58 provides decoding functionality with the FLICKR alphabet. BUT the library (as far as I know) only parses strings into vec<u8> (utf8). Flickr wants to calculate an integer value that depends on indexes. As Bs58 is not capable of it and the implementation is not that hard, I suggest to do it by our selfs.
(See https://www.dcode.fr/base-58-cipher {short id} -> into Integer)

Reply:
encodings like base58, base64 etc are ment to encode binary data as strings (and in reverse).
Thus you need to _decode_ the short url and get binary data back. This binary data you then have to interpret as a number (as flickr ids are numbers) and convert it into a string for our representation.

Reply:
for this conversion you have to figure out the byte ordering (Big endien or little endien) used